src/coreComponents/physicsSolvers/fluidFlow/FlowSolverBase.cpp

```diff
@@ -712,5 +712,56 @@ void FlowSolverBase::saveAquiferConvergedState( real64 const & time,
   } );
 }
 
+void FlowSolverBase::prepareWeights( DomainPartition & domain )
+{
+  forDiscretizationOnMeshTargets( domain.getMeshBodies(), [&] ( string const &,
+                                                                MeshLevel & mesh,
+                                                                arrayView1d< string const > const & regionNames )
+  {
+    GEOS_UNUSED_VAR( regionNames );
+
+    NumericalMethodsManager const & numericalMethodManager = domain.getNumericalMethodManager();
+    FiniteVolumeManager const & fvManager = numericalMethodManager.getFiniteVolumeManager();
+    FluxApproximationBase const & fluxApprox = fvManager.getFluxApproximation( getDiscretizationName() );
+    ElementRegionManager::ElementViewAccessor< arrayView1d< real64 const > > hydraulicAperture =
+      mesh.getElemManager().constructViewAccessor< array1d< real64 >, arrayView1d< real64 const > >( fields::flow::hydraulicAperture::key() );
+
+    fluxApprox.forStencils< SurfaceElementStencil, FaceElementToCellStencil, EmbeddedSurfaceToCellStencil >( mesh, [&]( auto & stencil )
+    {
+      typename TYPEOFREF( stencil ) ::KernelWrapper stencilWrapper = stencil.createKernelWrapper();
+
+      forAll< parallelHostPolicy >( stencilWrapper.size(), [=] ( localIndex const iconn )
```

Comment:
```suggestion
      forAll< parallelDevicePolicy<> >( stencilWrapper.size(), [=] ( localIndex const iconn )
```

Reply:
When I tried `parallelDevicePolicy<>` the GPU compilation didn't pass.

Reply:
This is because it's capturing `hydraulicAperture` by value, which is in fact something like `array1d< array1d< arrayView1d< real64 const > > >` and we can't copy arrays by value in kernels. You may want to turn it into a view
```c++
auto hydraulicApertureView = hydraulicAperture.toView();
```
and use that variable inside the RAJA loop.

Reply:
ah right, I had not noticed. Thanks @klevzoff

Reply:
I think it may be `toNestedView()`

Reply:
Thanks for the input. 
What type should I use for `hydraulicApertureView ` in the parameter list. 
Something like:
`LvArray::ArrayView<geos::array1d<geos::arrayView1d<const geos::real64>>, 1, 0, geos::localIndex, LvArray::ChaiBuffer>`
looks complicated 😊

Reply:
I think you can use:

`ElementRegionManager::ElementViewAccessor< arrayView1d< real64 const > >;`

Reply:
I get no viable conversion...
`FlowSolverBase.cpp:736:68: error: no viable conversion from 'const LvArray::ArrayView<const LvArray::ArrayView<const LvArray::ArrayView<const double, 1, 0, int, ChaiBuffer>, 1, 0, int, ChaiBuffer>, 1, 0, int, ChaiBuffer>' to 'ElementRegionManager::ElementViewAccessor<arrayView1d<const real64>>' (aka 'Array<LvArray::Array<LvArray::ArrayView<const double, 1, 0, int, ChaiBuffer>, 1, camp::int_seq<long, 0>, int, LvArray::ChaiBuffer>, 1, camp::int_seq<long, 0>, int, LvArray::ChaiBuffer>')
        stencilWrapper.removeHydraulicApertureContribution( iconn, hydraulicApertureView );`

I tried with `toView()`, `toNestedView()`, and `toNestedViewConst()`

Reply:
oh, sorry. ElementRegionManager::ElementViewConst<arrayView1d< real64 const >>

Reply:
oh, sorry, ElementRegionManager::ElementViewConst<arrayView1d< real64 const >>